public/utils/validation.ts

```diff
@@ -10,6 +10,12 @@ export const MAX_NAME_CHARACTERS = 50;
 // numbers 0-9, hyphens, spaces, and underscores.
 export const NAME_REGEX = new RegExp(/^[a-zA-Z0-9 _-]{5,50}$/);
 
+export const DETECTION_NAME_REGEX = new RegExp(/^[a-zA-Z0-9_]{5,50}$/);
+
+export const CONDITION_REGEX = new RegExp(
```

Comment:
Nitpick: Similar to what you have on line 10, it could be helpful to include an explanation of the regex here.

Reply:
added